test/e2e/istio_test.go

```diff
@@ -0,0 +1,212 @@
+	}
+	const vertxDeploymentName = "myapp"
+	err = WaitForDeployment(t, fw.KubeClient, namespace, vertxDeploymentName, 1, retryInterval, timeout)
+	require.NoError(t, err, "Failed waiting for myapp deployment")
+
+	// Add a liveliness probe to create some traces
+	vertxPort := intstr.IntOrString{IntVal: 8080}
+	livelinessHandler := &corev1.HTTPGetAction{Path: "/", Port: vertxPort, Scheme: corev1.URISchemeHTTP}
+	handler := &corev1.Handler{HTTPGet: livelinessHandler}
+	livelinessProbe := &corev1.Probe{Handler: *handler, InitialDelaySeconds: 1, FailureThreshold: 3, PeriodSeconds: 10, SuccessThreshold: 1, TimeoutSeconds: 1}
+
+	err = wait.Poll(retryInterval, timeout, func() (done bool, err error) {
+		vertxDeployment, err := fw.KubeClient.AppsV1().Deployments(namespace).Get(context.Background(), vertxDeploymentName, metav1.GetOptions{})
+		require.NoError(t, err)
+		containers := vertxDeployment.Spec.Template.Spec.Containers
+		for index, container := range containers {
+			if container.Name == vertxDeploymentName {
+				vertxDeployment.Spec.Template.Spec.Containers[index].LivenessProbe = livelinessProbe
+				updatedVertxDeployment, err := fw.KubeClient.AppsV1().Deployments(namespace).Update(context.Background(), vertxDeployment, metav1.UpdateOptions{})
+				if err != nil {
+					log.Warnf("Error %v updating vertx app, retrying", err)
+					return false, nil
+				}
+				log.Infof("Updated deployment %v", updatedVertxDeployment.Name)
+				return true, nil
+			}
+		}
+		return false, errors.New("Vertx deployment not found")
+	})
+	require.NoError(t, err)
+
+	err = wait.Poll(retryInterval, timeout, func() (done bool, err error) {
+		pods, err := fw.KubeClient.CoreV1().Pods(namespace).List(context.Background(), metav1.ListOptions{
+			LabelSelector: labels.FormatLabels(map[string]string{
+				"app": vertxDeploymentName,
+			}),
+			Limit: 10,
+		})
+		require.NoError(t, err)
+
+		if pods.Size() == 0 {
+			return false, errors.New("Vertx Pods not found")
+		}
+
+		for _, pod := range pods.Items {
+			exist := containerExistsInPod(pod.Spec.Containers, "istio-proxy")
+			if exist {
+				log.Infof("Istio-proxy found in pod %s", pod.Name)
+				return true, nil
+			} else {
+				return false, nil
+			}
+		}
+
+		return false, errors.New("Unexpected error while checking pods")
+	})
+
+	require.NoError(t, err, "Fail to wait for istio-proxy injection")
+
+	// Confirm that we've created some traces
+	ports := []string{"0:16686"}
+	portForward, closeChan := CreatePortForward(namespace, jaegerInstanceName, "all-in-one", ports, fw.KubeConfig)
+	defer portForward.Close()
+	defer close(closeChan)
+	forwardedPorts, err := portForward.GetPorts()
+	require.NoError(t, err)
+	queryPort := strconv.Itoa(int(forwardedPorts[0].Local))
+
+	url := "http://localhost:" + queryPort + "/api/traces?service=order"
+	err = WaitAndPollForHTTPResponse(url, func(response *http.Response) (bool, error) {
+		body, err := ioutil.ReadAll(response.Body)
+		if err != nil {
```

Comment:
Please replace this and the next "if err != nil" with require.equal

Reply:
Done. I guess you meant `require.NoError`.